tests/test_storage.py

```diff
@@ -634,17 +634,42 @@ def test_find_or_create_user_new(mocker):
     mock_session.commit.assert_called_once_with()
 
 
-def test_find_new_messages(mocker):
-    mock_session = mocker.MagicMock()
-    mock_submission = mocker.MagicMock()
-    mock_submission.is_downloaded = False
-    mock_submissions = [mock_submission]
-    mock_session.query().filter_by().all.return_value = mock_submissions
-    submissions = find_new_messages(mock_session)
-    assert submissions[0].is_downloaded is False
-
-
-def test_find_new_files(mocker):
+def test_find_new_messages(mocker, session):
+    source = factory.Source()
+    message_not_downloaded = factory.Message(
+        source=source,
+        is_downloaded=False,
+        is_decrypted=None,
+        content=None)
+    message_decrypt_not_attempted = factory.Message(
+        source=source,
+        is_downloaded=True,
+        is_decrypted=None,
+        content=None)
+    message_decrypt_failed = factory.Message(
+        source=source,
+        is_downloaded=True,
+        is_decrypted=False,
+        content=None)
+    message_decrypt_success = factory.Message(
+        source=source,
+        is_downloaded=True,
+        is_decrypted=True,
+        content='teehee')
+    session.add(source)
+    session.add(message_decrypt_not_attempted)
+    session.add(message_not_downloaded)
+    session.add(message_decrypt_failed)
+    session.add(message_decrypt_success)
```

Comment:
I feel like for completeness and future proofing, we may want to add a `session.commit()` here. In part because all the DB objects don't get an `id` set until after a `flush()` or `commit()`. It will also force all the integrity checks we have baked in to SQLite to execute.